Core.DomainModel/Organization/StsOrganizationConnection.cs

```diff
@@ -9,30 +12,81 @@ namespace Core.DomainModel.Organization
         /// </summary>
         public int? SynchronizationDepth { get; set; }
+        public virtual ICollection<StsOrganizationChangeLog> StsOrganizationChangeLogs { get; set; }
+
 
         public bool SubscribeToUpdates { get; set; }
-        //TODO https://os2web.atlassian.net/browse/KITOSUDV-3317 adds the change logs here
+        
         public DisconnectOrganizationFromOriginResult Disconnect()
         {
             var organizationUnits = Organization.OrgUnits.Where(x => x.Origin == OrganizationUnitOrigin.STS_Organisation).ToList();
             organizationUnits.ForEach(unit => unit.ConvertToNativeKitosUnit());
+            var removedLogs = RemoveAllLogs();
 
             Connected = false;
             SubscribeToUpdates = false;
             SynchronizationDepth = null;
-            return new DisconnectOrganizationFromOriginResult(organizationUnits);
+            return new DisconnectOrganizationFromOriginResult(organizationUnits, removedLogs);
         }
 
         public IExternalOrganizationalHierarchyUpdateStrategy GetUpdateStrategy()
         {
             return new StsOrganizationalHierarchyUpdateStrategy(Organization);
         }
+
+        public StsOrganizationConnectionImportLogResult AddNewLogs(IEnumerable<StsOrganizationChangeLog> newLogs)
+        {
+            var newLogsList = newLogs.ToList();
+            foreach (var newLog in newLogsList)
+            {
+                StsOrganizationChangeLogs.Add(newLog);
+            }
+
+            return RemoveOldestLogs(newLogsList);
+        }
+
+        public Result<IEnumerable<StsOrganizationChangeLog>, OperationError> GetLastNumberOfChangeLogs(int number = StsOrganizationConnectionConstants.TotalNumberOfLogs)
+        {
+            if (number <= 0)
+            {
+                return new OperationError("Number of change logs to get cannot be lower than 0", OperationFailure.BadInput);
+            }
+
+            return StsOrganizationChangeLogs
+                .OrderByDescending(x => x.LogTime)
+                .Take(number)
+                .ToList();
+        }
+
+        private IEnumerable<StsOrganizationChangeLog> RemoveAllLogs()
+        {
+            var changeLogs = StsOrganizationChangeLogs.ToList();
+            foreach (var changeLog in changeLogs)
+            {
+                StsOrganizationChangeLogs.Remove(changeLog);
+            }
+
+            return changeLogs;
+        }
+
+        private StsOrganizationConnectionImportLogResult RemoveOldestLogs(IEnumerable<StsOrganizationChangeLog> newLogs)
+        {
+            var logsToRemove = StsOrganizationChangeLogs.OrderByDescending(x => x.LogTime).Skip(StsOrganizationConnectionConstants.TotalNumberOfLogs).ToList();
+            logsToRemove.ForEach(log => StsOrganizationChangeLogs.Remove(log));
+
+            return new StsOrganizationConnectionImportLogResult(newLogs, logsToRemove);
```

Comment:
you are combining newLogs which is from a different use case with something related to the removal case.. return type is messy